### DISCUSSION TITLE
Displaying a Vec<String>

### DISCUSSION THREAD

**Discussion by pieterdd** [slint-ui/slint#4632] [Q&A] [answered]
I'm quite new to Slint. Let's say I have a `Vec<String>` in my Rust code. What's the most straightforward way to inject it into a Slint window and have the window automatically react to changes in the vector?

The closest [example I've seen so far](https://docs.rs/slint/latest/slint/struct.ModelRc.html#example) is actually a `Vec<SharedString>` (which I suppose I could map the vector items to) that's used to create a reference-counted `VecModel` which is used to create a `ModelRc`. Is this the best available option?

**Comment by ogoffart** [comment, marked as answer]
Right, you will want to map the Vec<String> to a Vec<SharedString>.
You can do something like so:

```rust
let my_vec : Vec<String> = ...;
let my_vec : Vec<slint::SharedString> = my_vec.into_iter().map(Into::into).collect();
let model = slint::ModelRc::new(VecModel::from(vec!["Hello".into(), "World".into()]));

my_ui.set_some_property(model);
```

(I haven't tried this code)

If you want to automatically react to change in the vector, you may want to have a `Rc<VecModel<SharedString>>` instead and operate in that instead of on the `Vec<String>`

**Reply by pieterdd** [reply]
In that case I'll update my data model to use slint::SharedString instead of String, keeping type conversions and other friction to a minimum. Thanks for the advice Olivier!
